### PR TITLE
fix: remove requirement for URL to end with .zip

### LINKED ISSUE #3853: zip starter projects should not require URLs ending with .zip
/kind bug

## What versions of software are you using?

**Operating System:** macOS 10.15.6

**Output of `odo version`:** odo v1.2.5 (0780a6bbf)

## How did you run odo exactly?
`odo create --devfile ../devfile.yaml --starter=foo` with a devfile I created explicitly to have several starter zip projects

## Actual behavior
 ✗  Invalid zip url: <url that doesn't end with .zip>

## Expected behavior
If the URL points to a non-zip file, it should only fail when trying to unzip it, not prevent people from using URLs that don't end in `.zip`

## Any logs, error output, etc?
See above.

### PR DESCRIPTION
**Pull request by metacosm**
**What type of PR is this?**
/kind bug

**Which issue(s) this PR fixes**:

Fixes #3853

### CODE REVIEW

File: pkg/util/util.go [hunk 1]
```diff
@@ -904,9 +904,6 @@ func GetAndExtractZip(zipURL string, destination string, pathToUnzip string) err
 	if zipURL == "" {
 		return errors.Errorf("Empty zip url: %s", zipURL)
 	}
-	if !strings.Contains(zipURL, ".zip") {
-		return errors.Errorf("Invalid zip url: %s", zipURL)
-	}
 
 	var pathToZip string
 	if strings.HasPrefix(zipURL, "file://") {
```
Comment (girishramnani) [inline pkg/util/util.go:909]: I am ok with this PR but do we want to also enforce that the url starts with `file://` a url could be a simple http or https one but could have a zip file returned?

Comment (metacosm) [reply pkg/util/util.go:909]: You're missing the rest of the code which deals with that… 😉

**Comment by amitkrout** [comment]
platform issue
/test v4.5-integration-e2e

**Comment by kadel** [comment]
small fix
/lgtm
/approve

**Pull request merged**